core/recorder.cc

```diff
@@ -0,0 +1,93 @@
+/** recorder.cc
+    Jeremy Barnes, 26 March 2016
+    Copyright (c) 2016 Datacratic Inc.  All rights reserved.
+*/
+
+#include "recorder.h"
+
+namespace Datacratic {
+namespace MLDB {
+
+
+/*****************************************************************************/
```

Comment:
Yes.  They are copied on the Destructive version, as this allows the calling code to construct the objects which can then be moved into place by the implementation, and gives the possibility for the calling code to move them directly if it can.

The non-Destructive versions take references.